api/app/tests/apps/email_integrations/test_mail.py

```diff
@@ -356,10 +381,71 @@ def test_links_environment_env_var_not_set(self):
         # we want a history entry when a email was sent
         self.assertEqual(Note.objects.count(), len(env_fe_mapping))
 
+    def test_send_mail_reporter_created_send_mail_false(self):
+        # Is the intended rule activated?
+        actions = self._get_mail_rules(['Send mail signal created'])._get_actions(self.signal)
+        self.assertEqual(len(actions), 1)
+
+        ma = MailActions(mail_rules=SIGNAL_MAIL_RULES)
+        activated = ma._get_actions(self.signal)
+        self.assertEqual(set(actions), set(activated))
+
+        # Check mail contents
+        ma.apply(signal_id=self.signal.id, send_mail=False)
+
+        # No mail should be sent
+        self.assertEqual(len(mail.outbox), 0)
+
+        # we want a history entry when a email was sent
+        self.assertEqual(Note.objects.count(), 0)
+
+    def test_send_mail_reporter_created_no_note(self):
+        # Is the intended rule activated?
+        actions = self._get_mail_rules(['Send mail signal created'])._get_actions(self.signal)
+        self.assertEqual(len(actions), 1)
+
+        # We do not want to add a note so we should remove the 'history_entry_text'
+        COPIED_SIGNAL_MAIL_RULES = [copy.deepcopy(r)
+                                    for r in SIGNAL_MAIL_RULES if r['name'] == 'Send mail signal created']
+        COPIED_SIGNAL_MAIL_RULES[0]['additional_info'].pop('history_entry_text')
 
-class TestOptionalMails(TestCase):
+        # Is it the only one that activates?
+        ma = MailActions(mail_rules=COPIED_SIGNAL_MAIL_RULES)
+        activated = ma._get_actions(self.signal)
+        self.assertEqual(set(actions), set(activated))
+
+        # Check mail contents
+        ma.apply(signal_id=self.signal.id)
+        self.assertEqual(len(mail.outbox), 1)
+
+        # we want a history entry when a email was sent
+        self.assertEqual(Note.objects.count(), 0)
+
+    def test_send_mail_reporter_created_no_additional_context(self):
+        # Is the intended rule activated?
+        actions = self._get_mail_rules(['Send mail signal created'])._get_actions(self.signal)
+        self.assertEqual(len(actions), 1)
+
+        # We do not want to add a note so we should remove the 'history_entry_text'
+        COPIED_SIGNAL_MAIL_RULES = [copy.deepcopy(r)
+                                    for r in SIGNAL_MAIL_RULES if r['name'] == 'Send mail signal created']
+        COPIED_SIGNAL_MAIL_RULES[0]['kwargs'].pop('context')
+
+        # Is it the only one that activates?
+        ma = MailActions(mail_rules=COPIED_SIGNAL_MAIL_RULES)
+        activated = ma._get_actions(self.signal)
+        self.assertEqual(set(actions), set(activated))
+
+        # Check mail contents
+        ma.apply(signal_id=self.signal.id)
+        self.assertEqual(len(mail.outbox), 1)
```

Comment:
Should we check the email content to check that the extra context stuff is not there?